releasenotes/notes/fix-instruction-c_if-3334bc8bcc38a327.yaml

```diff
@@ -0,0 +1,15 @@
+---
+deprecations:
+  - |
+    Creating an instance of :obj:`.InstructionSet` with the ``circuit_cregs`` keyword argument is deprecated.
```

Comment:
The leading `.` means search the symbol table for a symbol with this suffix, and then display only the suffix.  In this case it resolves fully to `qiskit.circuit.InstructionSet`.  For dot access, the search order goes from "nearest" to "furthest" roughly, so if I reference `.InstructionSet` in `qiskit.a.b`, it'll search
1. `qiskit.a.b.InstructionSet`
2. `qiskit.a.InstructionSet`
3. `qiskit.InstructionSet`
4. `InstructionSet`
5. _anything_`.InstructionSet`
I think it's meant to warn if there are duplicate candidates for resolution, but I usually only use the shorthand when I'm fairly sure there's only one thing it could match anyway, because otherwise it's likely to be confusing in the text as well.

`:obj:` is magic and resolves to whatever role the name is first found in, so it can match `:class:`, `:meth:`, `:attr:`, etc, but if the name exists in more than one role, you can't generally predict which you'll get.  In this case, there's only one `InstructionSet`, so it'll

Reply:
Awesome, that's super handy to know!